src/StackExchange.Exceptional.AspNetCore/ExceptionalLogger.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace StackExchange.Exceptional
+{
+    class ExceptionalLogger : ILogger
+    {
+        private readonly string _category;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IOptions<ExceptionalSettings> _settings;
+        private readonly bool _ignored;
+        private static readonly HashSet<string> _ignoredCategories = new HashSet<string>
+        {
+            typeof(ExceptionalMiddleware).FullName, // exceptional middleware calls some ILogger stuff itself, ignore those calls)
+        };
+
+        public ExceptionalLogger(string category, IOptions<ExceptionalSettings> settings, IHttpContextAccessor httpContextAccessor = null)
+        {
+            _category = category;
+            _settings = settings;
+            _httpContextAccessor = httpContextAccessor;
+            _ignored = _ignoredCategories.Contains(category);
```

Comment:
Alternatively, we could attach something to Exception.Data, to prevent the same exception from being logged twice.

A reasonable case of this happening would be if an exception gets logged to an ILogger, is then rethrown, is ultimately captured by the middleware, and is logged again...

Reply:
I ran into exactly that while doing my own kinda version of this and doing that worked great:

```csharp
/// <summary>
/// This is used internally to stop an exception from being logged twice
/// <para />
/// This is usually needed when code closer to the source logs extra information e.g. ids
/// and then rethrows to trigger the error middleware, which ends up logging it again
/// <para />
/// Returns true if the exception has already been logged, or false otherwise
/// </summary>
/// <param name="exception"></param>
public static bool ExceptionLogged(Exception exception)
{
    var key = $"Exceptional.Logged";
    var logged = exception.Data.Contains(key);
    if (!logged)
    {
        exception.Data[key] = true;
    }
    return logged;
}
```